consensus/view_change.go

```diff
@@ -278,9 +278,17 @@ func (consensus *Consensus) startNewView(viewID uint64, newLeaderPriKey *bls.Pri
 	consensus.mutex.Lock()
 	defer consensus.mutex.Unlock()
 
+	if !consensus.IsViewChangingMode() {
```

Comment:
Can we move this piece of code to the start of `onViewChange(consensus/view_change.go:317)`?

Reply:
That won't help.  The new leader may still in the viewchanging mode, when it received two view change messages at the same time.  However, one is processed and ready to start a new view, another one is still being processed and just came a bit late.  the OnViewChange is not protected by mutex.  However, startNewView should only be called once. That's why the check at the top of this function, since this function is protected by mutex. That's guaranteed to be called only once per viewchange.

Reply:
Fair. Thanks for clarification